src/app/views/provisioning/subscriptions.controller.coffee

```diff
@@ -37,25 +37,33 @@ angular.module 'mnoEnterpriseAngular'
           $state.go("home.subscriptions", {subType: 'active'})
       )
 
-    orgPromise = MnoeOrganizations.get()
-    subPromise = vm.subscriptionsPromise()
-
-    $q.all({organization: orgPromise, subscriptions: subPromise}).then(
-      (response) ->
-        vm.subscriptions = response.subscriptions
-        if vm.cartSubscriptions && vm.subscriptions.length < 1
-          toastr.info('mno_enterprise.templates.dashboard.provisioning.subscriptions.cart.empty')
-          $state.go('home.marketplace')
-          return
-
-        vm.orgCurrency = response.organization.organization?.billing_currency || MnoeConfig.marketplaceCurrency()
-
-        # If a subscription doesn't contains a pricing for the org currency, a warning message is displayed
-        vm.displayCurrencyWarning = not _.every(response.subscriptions, (subscription) ->
-          currencies = _.map(subscription?.product_pricing?.prices, 'currency')
-          _.includes(currencies, vm.orgCurrency) || (subscription?.product_pricing?.pricing_type in PRICING_TYPES['unpriced'])
-        )
-    ).finally(-> vm.isLoading = false)
+    vm.initialize = ->
+      vm.isLoading = true
+      orgPromise = MnoeOrganizations.get()
+      subPromise = vm.subscriptionsPromise()
+
+      $q.all({organization: orgPromise, subscriptions: subPromise}).then(
+        (response) ->
+          vm.subscriptions = response.subscriptions
+          if vm.cartSubscriptions && vm.subscriptions.length < 1
+            toastr.info('mno_enterprise.templates.dashboard.provisioning.subscriptions.cart.empty')
+            $state.go('home.marketplace')
+            return
+
+          vm.orgCurrency = response.organization.organization?.billing_currency || MnoeConfig.marketplaceCurrency()
+
+          # If a subscription doesn't contains a pricing for the org currency, a warning message is displayed
+          vm.displayCurrencyWarning = not _.every(response.subscriptions, (subscription) ->
+            currencies = _.map(subscription?.product_pricing?.prices, 'currency')
+            _.includes(currencies, vm.orgCurrency) || (subscription?.product_pricing?.pricing_type in PRICING_TYPES['unpriced'])
+          )
+      ).finally(-> vm.isLoading = false)
+
+    #====================================
+    # Post-Initialization
+    #====================================
+    $scope.$watch MnoeOrganizations.getSelectedId, (val) ->
+      vm.initialize() if val?
```

Comment:
Can we also reset the UI loader here. Otherwise screen might remain frozen till the subscriptions are re-fetched for the new organization.

Reply:
Sorry nevermind, I see the loader is set in the `initialize` method